homeassistant/components/zwave_js/sensor.py

```diff
@@ -47,6 +84,124 @@
 LOGGER = logging.getLogger(__name__)
 
 
+@dataclass
+class ZwaveSensorBaseEntityDescription(SensorEntityDescription):
+    """Base description of a ZwaveSensorBase entity."""
+
+    info: ZwaveDiscoveryInfo | None = None
+
+    def __post_init__(self) -> None:
+        """Post initialize."""
+        self.force_update = True
+
+        if (device_class := self._device_class) is not None:
+            self.device_class = device_class
+        if (state_class := self._state_class) is not None:
+            self.state_class = state_class
+
+    @property
+    def _device_class(self) -> str | None:
+        """Return the device class."""
+        return None
+
+    @property
+    def _state_class(self) -> str | None:
+        """Return the state class."""
+        return None
+
+
+@dataclass
+class ZwaveMeterSensorEntityDescription(ZwaveSensorBaseEntityDescription):
+    """Description of a Z-Wave Meter CC entity."""
+
+    meter_type: MeterType = field(init=False)
+    scale_type: IntEnum = field(init=False)
+
+    def __post_init__(self) -> None:
+        """Post initialize."""
+        assert self.info
+        cc_specific = self.info.primary_value.metadata.cc_specific
+        meter_type_id = cc_specific[CC_SPECIFIC_METER_TYPE]
+        scale_type_id = cc_specific[CC_SPECIFIC_SCALE]
+        self.meter_type = MeterType(meter_type_id)
```

Comment:
We don't seem to use `meter_type` outside post init?

Reply:
I wanted to add this as a state attribute because the reset meter service can take the meter_type as input, but `EntityDescription` doesn't support extra state attributes yet. I plan to submit a PR to add that

Reply:
Alternatively it just becomes something I can grab from the entity description since we have a separate class for Meter sensors